presto-main/src/main/java/com/facebook/presto/operator/aggregation/fixedhistogram/FixedDoubleBreakdownHistogram.java

```diff
@@ -87,52 +90,47 @@ public BucketWeight(double left, double right, double weight, long count)
 
     public FixedDoubleBreakdownHistogram(int bucketCount, double min, double max)
     {
+        verifyParameters(bucketCount, min, max);
         this.bucketCount = bucketCount;
         this.min = min;
         this.max = max;
-        verifyParameters(bucketCount, min, max);
         this.indices = new int[0];
         this.weights = new double[0];
         this.counts = new long[0];
     }
 
-    public FixedDoubleBreakdownHistogram(SliceInput input)
-    {
-        this.bucketCount = input.readInt();
-        this.min = input.readDouble();
-        this.max = input.readDouble();
-        verifyParameters(bucketCount, min, max);
-        int size = input.readInt();
-        this.indices = new int[size];
-        this.weights = new double[size];
-        this.counts = new long[size];
-        input.readBytes(
-                Slices.wrappedIntArray(indices),
-                size * SizeOf.SIZE_OF_INT);
-        input.readBytes(
-                Slices.wrappedDoubleArray(weights),
-                size * SizeOf.SIZE_OF_DOUBLE);
-        input.readBytes(
-                Slices.wrappedLongArray(counts),
-                size * SizeOf.SIZE_OF_LONG);
-    }
-
-    protected FixedDoubleBreakdownHistogram(FixedDoubleBreakdownHistogram other)
+    private FixedDoubleBreakdownHistogram(FixedDoubleBreakdownHistogram other)
     {
         this.bucketCount = other.bucketCount;
         this.min = other.min;
         this.max = other.max;
-        verifyParameters(bucketCount, min, max);
         this.indices = new int[other.indices.length];
         this.weights = new double[other.weights.length];
         this.counts = new long[other.counts.length];
-        for (int i = 0; i < other.indices.length; ++i) {
+        for (int i = 0; i < other.indices.length; i++) {
```

Comment:
Use Arrays.copyOf: `this.indices = Arrays.copyOf(other.indices, other.indices.length);`

Reply:
Using the `copyOf` version, thanks. 

As to the `++i`, am searching for it in the PR codebase, but it appears several times in the Presto codebase, BTW:

atavory-mbp:presto atavory$ find . -name '*.java' | xargs grep '++i' | wc
     118    1321   16659
atavory-mbp:presto atavory$ find . -name '*.java' | xargs grep '++j' | wc
       1      11     143